ext/spl/spl_directory.c

```diff
@@ -2045,7 +2045,7 @@ static int spl_filesystem_file_read(spl_filesystem_object *intern, int silent) /
 		intern->u.file.current_line = estrdup("");
 		intern->u.file.current_line_len = 0;
 	} else {
-		if (SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_DROP_NEW_LINE)) {
+		if (SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_DROP_NEW_LINE) && !SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_READ_CSV)) {
```

Comment:
Wouldn't this also affect the behavior of the `fgets()` method?

Reply:
Right! (in `case ::READ_CSV` is set)

Reply:
And the `!SplFileObject::READ_CSV` case is an issue, since `::fgetcsv()` exhibits the same broken behavior (namely, that it drops the first linebreak in a quoted field): https://3v4l.org/OEiFq. This should now be fixed.